rust/src/ortc.rs

```diff
@@ -641,29 +641,40 @@ pub(crate) fn can_consume(
 /// codecs' RTCP feedback and header extensions, and also enables or disabled RTX.
 #[allow(clippy::suspicious_operation_groupings)]
 pub(crate) fn get_consumer_rtp_parameters(
-    consumable_params: &RtpParameters,
-    caps: &RtpCapabilities,
+    consumable_rtp_parameters: &RtpParameters,
+    remote_rtp_capabilities: &RtpCapabilities,
     pipe: bool,
+    enable_rtx: bool,
 ) -> Result<RtpParameters, ConsumerRtpParametersError> {
     let mut consumer_params = RtpParameters {
-        rtcp: consumable_params.rtcp.clone(),
+        rtcp: consumable_rtp_parameters.rtcp.clone(),
         ..RtpParameters::default()
     };
 
-    for cap_codec in &caps.codecs {
+    for cap_codec in &remote_rtp_capabilities.codecs {
         validate_rtp_codec_capability(cap_codec)
             .map_err(ConsumerRtpParametersError::InvalidCapabilities)?;
     }
 
     let mut rtx_supported = false;
 
-    for mut codec in consumable_params.codecs.clone() {
-        if let Some(matched_cap_codec) = caps
+    for mut codec in consumable_rtp_parameters.codecs.clone() {
+        if !enable_rtx && codec.is_rtx() {
+            continue;
+        }
+
+        if let Some(matched_cap_codec) = remote_rtp_capabilities
             .codecs
             .iter()
             .find(|cap_codec| match_codecs(cap_codec.deref().into(), (&codec).into(), true).is_ok())
         {
-            *codec.rtcp_feedback_mut() = matched_cap_codec.rtcp_feedback().clone();
+            *codec.rtcp_feedback_mut() = matched_cap_codec
+                .rtcp_feedback()
+                .iter()
+                .filter(|&&fb| enable_rtx || fb != RtcpFeedback::Nack)
+                .copied()
+                .collect();
```

Comment:
- Why is iter() needed?
- Why &&fb?
- What does copies() do?
- And collect()

Reply:
- iter() - create iterator over &T in collection
- && - just double dereference, because T is &A, and after iter() - &&A
- copies() - map iterator with copy() call on each element(&T before copy and T after copy)
- collect() - create collection from iterator. In current case: collect::<Vec<_>>()

Reply:
Thanks. Just a question: why is copies() needed?

Reply:
Before copied() we have iterator over &T. If we collect without copy of elements, we receive Vec<&T>. This vector contains references to elements from source collection. After that we cannot assign Vec<&T> to Vec<T> variable.

Reply:
Thanks, so copied() receives a reference to a "struct", copies the full struct and returns it.

Reply:
Yes. For `copied()` type `T` should have trait `Copy`. In C++ terms: type T should have copy-constructor.
https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.copied